vicinity/backends/usearch.py

```diff
@@ -0,0 +1,191 @@
+        arguments: UsearchArgs,
+    ) -> None:
+        """Initialize the backend using Usearch."""
+        super().__init__(arguments)
+        self.index = index
+        self.keys: list[Any] = []
+        self.key_to_index: dict[int, int] = {}
+
+    @classmethod
+    def from_vectors(
+        cls: type[UsearchBackend],
+        vectors: npt.NDArray,
+        metric: Literal["cos", "ip", "l2sq", "hamming", "tanimoto"],
+        connectivity: int,
+        expansion_add: int,
+        expansion_search: int,
+        **kwargs: Any,
+    ) -> UsearchBackend:
+        """
+        Create a new instance from vectors.
+
+        :param vectors: The vectors to index.
+        :param metric: The metric to use.
+        :param connectivity: The connectivity parameter.
+        :param expansion_add: The expansion add parameter.
+        :param expansion_search: The expansion search parameter.
+        :param **kwargs: Additional keyword arguments.
+        :return: A new instance of the backend.
+        :raises TypeError: If the type of keys is not supported.
+        """
+        dim = vectors.shape[1]
+        index = UsearchIndex(
+            ndim=dim,
+            metric=metric,
+            connectivity=connectivity,
+            expansion_add=expansion_add,
+            expansion_search=expansion_search,
+        )
+        keys = index.add(keys=None, vectors=vectors)  # type: ignore
+        arguments = UsearchArgs(
+            dim=dim,
+            metric=metric,
+            connectivity=connectivity,
+            expansion_add=expansion_add,
+            expansion_search=expansion_search,
+        )
+        backend = cls(index, arguments=arguments)
+        if isinstance(keys, np.ndarray):
+            backend.keys = keys.tolist()
+        else:
+            raise TypeError(f"Unexpected type for keys: {type(keys)}")
+        backend.key_to_index = {key: idx for idx, key in enumerate(backend.keys)}
+
+        return backend
+
+    @property
+    def backend_type(self) -> Backend:
+        """The type of the backend."""
+        return Backend.USEARCH
+
+    @property
+    def dim(self) -> int:
+        """Get the dimension of the space."""
+        return self.index.ndim
+
+    def __len__(self) -> int:
+        """Get the number of vectors."""
+        return len(self.index)
+
+    @classmethod
+    def load(cls: type[UsearchBackend], base_path: Path) -> UsearchBackend:
+        """Load the index from a path."""
+        path = Path(base_path) / "index.usearch"
+        arguments = UsearchArgs.load(base_path / "arguments.json")
+        index = UsearchIndex(
+            ndim=arguments.dim,
+            metric=arguments.metric,
+            connectivity=arguments.connectivity,
+            expansion_add=arguments.expansion_add,
+            expansion_search=arguments.expansion_search,
+        )
+        index.load(str(path))
+        return cls(index, arguments=arguments)
+
+    def save(self, base_path: Path) -> None:
```

Comment:
I don't think you save or load the mapping, so once you load this it probably breaks?